src/main/java/Duke.java

```diff
@@ -1,10 +1,292 @@
+                           + System.lineSeparator()
+                           + INDENT + "What can I do for you?");
+    }
+
+    // parse user input in REPL loop
+    public static void parseUserInput() {
+        Scanner line = new Scanner(System.in);
+        String input = line.nextLine();
+
+        String userKeyWord;
+
+        while(!input.toLowerCase().equals("bye")) {
+            String[] inputArr = input.split(" ");
+
+            userKeyWord = inputArr[0].toLowerCase();
+
+            switch (userKeyWord) {
+            case("todo"):
+            case("deadline"):
+            case("event"): {
+                addTask(inputArr);
+                break;
+            }
+            case("delete"): {
+                try {
+                    deleteTask(Integer.parseInt(inputArr[1]));
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    System.out.println(INDENT + "Please input a number!");
+                } catch (NumberFormatException e) {
+                    System.out.println(INDENT + "Sorry, that is not a valid input!");
+                }
+                break;
+            }
+            case("done"): {
+                try {
+                    markTaskAsDone(Integer.parseInt(inputArr[1]));
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    System.out.println(INDENT + "Please input a number!");
+                } catch (NumberFormatException e) {
+                    System.out.println(INDENT + "Sorry, that is not a valid input!");
+                }
+                break;
+            }
+            case("list"): {
+                printAllTasks();
+                break;
+            }
+            default: {
+                System.out.println(INDENT + "Oops, I don't understand! D:");
+            }
+            }
+            input = line.nextLine();
+        }
+    }
+
+    // shut down Duke
+    public static void shutDownProgram() {
+        try {
+            writeToFile(dataFilePath);
+        } catch (IOException e) {
+            System.out.println(INDENT + "Shoots, an error occurred while saving your list data. :(");
+        }
+        System.out.println(INDENT + "Bye. Hope to see you again soon!");
+    }
+
+    // read previously stored task data from duke.txt file, when program first starts up
+    public static void readFileContents(String filePath) throws FileNotFoundException {
+        FileReader fr = new FileReader(filePath);
+        Scanner s = new Scanner(fr);
+        while (s.hasNext()) {
+            String[] taskStrArr = s.nextLine().split(">");
+//            for (int i = 0; i < taskStrArr.length; i++) {
+//                taskStrArr[i] = taskStrArr[i].trim();
+//            }
+
+            // strArr[0] - type of Task (T/D/E)
+            // strArr[1] - whether task isDone (0/1)
+            // strArr[2] - task description
+            // strArr[3] - by/at (for deadline/event only)
```

Comment:
You can consider removing old code from the codebase instead of commenting it when making a PR